src/main/scala/com/fulcrumgenomics/util/ProgressLogger.scala

```diff
@@ -42,9 +43,48 @@ case class ProgressLogger(logger: Logger,
   /** Method to use to record progress when genome location isn't available or relevant. */
   def record(): Boolean = record(null, -1)
 
-  def record(rec: SamRecord): Boolean = record(rec.asSam)
+  def record(rec: SamRecord): Boolean = super.record(rec.asSam)
+
+  def record(variant: Variant): Boolean = record(variant.chrom, variant.pos)
+
+  def record(interval: Interval): Boolean = record(interval.getContig, interval.getStart)
+
+  /** Record progress of a single item.
+    *
+    * @param item the item log
+    * @param fallback the fallback method to record progress if the type is not a type supported by [[ProgressLogger]].
+    * @return
+    */
+  def record[ItemType](item: ItemType,
+                       fallback: (ProgressLogger, ItemType) => Boolean = (p: ProgressLogger, _: ItemType) => p.record()): Boolean = {
```

Comment:
If you don't like this method on `ProgressLogger` but like the `ProgressLoggingIterator`, I can move the pattern matching directly into the method in `ProgressLoggingIterator`.

Reply:
I think it would be better to have this be a private def in the iterator as you suggest.